x-pack/plugins/security_solution/public/resolver/view/resolver_loading_state.test.tsx

```diff
@@ -0,0 +1,173 @@
+        dataAccessLayer,
+        databaseDocumentID,
+        resolverComponentInstanceID,
+      });
+    });
+
+    it('should display a loading state', async () => {
+      // Trigger a loading state by requesting data based on a new DocumentID.
+      // There really is no way to do this in the view besides changing the url, so triggering the action instead
+      await expect(
+        simulator.map(() => ({
+          resolverGraphLoading: simulator.testSubject('resolver:graph:loading').length,
+          resolverGraphError: simulator.testSubject('resolver:graph:error').length,
+          resolverGraph: simulator.testSubject('resolver:graph').length,
+        }))
+      ).toYieldEqualTo({
+        resolverGraphLoading: 1,
+        resolverGraphError: 0,
+        resolverGraph: 0,
+      });
+    });
+  });
+
+  describe('When resolver tree data is being requested', () => {
+    beforeEach(() => {
+      const {
+        metadata: { databaseDocumentID },
+        dataAccessLayer,
+        pause,
+        resume,
+      } = pausableNoAncestorsTwoChildren();
+      pause({ resolverTree: true });
+      simulator = new Simulator({
+        dataAccessLayer,
+        databaseDocumentID,
+        resolverComponentInstanceID,
+      });
+    });
+
+    it('should display a loading state', async () => {
+      // Trigger a loading state by requesting data based on a new DocumentID.
+      // There really is no way to do this in the view besides changing the url, so triggering the action instead
+      await expect(
+        simulator.map(() => ({
+          resolverGraphLoading: simulator.testSubject('resolver:graph:loading').length,
+          resolverGraphError: simulator.testSubject('resolver:graph:error').length,
+          resolverGraph: simulator.testSubject('resolver:graph').length,
+        }))
+      ).toYieldEqualTo({
+        resolverGraphLoading: 1,
+        resolverGraphError: 0,
+        resolverGraph: 0,
+      });
+    });
+  });
+
+  describe("When the entities request doesn't return any data", () => {
+    beforeEach(() => {
+      const {
+        metadata: { databaseDocumentID },
+        dataAccessLayer,
+      } = emptiableNoAncestorsTwoChildren({ entities: true });
+
+      simulator = new Simulator({
+        dataAccessLayer,
+        databaseDocumentID,
+        resolverComponentInstanceID,
+      });
+    });
+
+    it('should display an error', async () => {
+      await expect(
+        simulator.map(() => ({
+          resolverGraphLoading: simulator.testSubject('resolver:graph:loading').length,
+          resolverGraphError: simulator.testSubject('resolver:graph:error').length,
+          resolverGraph: simulator.testSubject('resolver:graph').length,
+        }))
+      ).toYieldEqualTo({
+        resolverGraphLoading: 0,
+        resolverGraphError: 1,
+        resolverGraph: 0,
+      });
+    });
+  });
+
+  describe("When the resolver tree request doesn't return any data", () => {
```

Comment:
This should really never happen since the `resolverTree` request only takes place after successfully getting an `entityID`. Have this test here though so we're at least checking the behavior in case something wonky happens.

Reply:
This could happen if different indices are used for `entities` than what are used for `resolverTree`. I think that might even be the case today.